src/components/boolean-search/boolean-table.tsx

```diff
@@ -0,0 +1,307 @@
+        // Map your rows data here
+        and: index === 0 ? '' : 'AND',
+        strength: (
+          <select
+            value={row.strength || defaultValues.strength}
+            onChange={(e) => handleChange(index, 'strength', e.target.value)}
+          >
+            {strength?.map((option: any, index: any) => {
+              return <option key={index}>{option.Strength}</option>;
+            })}
+          </select>
+        ),
+        source: (
+          <>
+            <input
+              className='red-text'
+              value={row.source.name || ''}
+              disabled
+              required
+              onChange={(e) => handleChange(index, 'source', e.target.value)}
+            />
+            {row.sourceInvalid ? <div className='red-text'>Please enter a value for this field.</div> : <></>}
+          </>
+        ),
+        stages: (
+          <>
+            <span>From: </span>
+            <select
+              className='stageDropdown'
+              value={row.stageFrom.Name || defaultValues.stageFrom.Name}
+              onChange={(e) => handleChange(index, 'stageFrom', e.target.value)}
+            >
+              {stageFrom?.map((option: any, index: any) => {
+                return <option key={index}>{option.Name}</option>;
+              })}
+            </select>
+            <span>To: </span>
+            <select
+              className='stageDropdown'
+              value={row.stageTo.Name || defaultValues.stageTo.Name}
+              onChange={(e) => handleChange(index, 'stageTo', e.target.value)}
+            >
+              {row.toStageOptions?.map((option: any, index: any) => {
+                return <option key={index}>{option.Name}</option>;
+              })}
+            </select>
+          </>
+        ),
+        pattern: (
+          <select
+            value={row.pattern || defaultValues.pattern || ''}
+            onChange={(e) => handleChange(index, 'pattern', e.target.value)}
+          >
+            <option style={{ display: 'none' }} disabled selected value=''></option>
+            {pattern?.map((option: any, index: any) => (
+              <option key={index}>{option.Pattern}</option>
+            ))}
+          </select>
+        ),
+        location: (
+          <select
+            value={row.location || defaultValues.location || ''}
+            onChange={(e) => handleChange(index, 'location', e.target.value)}
+          >
+            <option style={{ display: 'none' }} disabled selected value=''></option>
+            {location?.map((option: any, index: any) => {
+              return (
+                <option key={index}>
+                  {option.Pattern_Location}
+                </option>
+              );
+            })}
+          </select>
+        ),
+        actions: <button onClick={(e) => handleRemoveRow(e, index)}>Remove</button>
+      })),
+    [rows, strength, pattern, location, stageFrom, source]
+  );
+
+  const tableInstance = useTable({
```

Comment:
Please explain what the benefits of using this react library are.

It seems like a lot of extra information to learn/understand for something that can be defined ourself. The other concern is about support for this library. I don't expect it to change, but if something does that means more overhead for us to come back and fix this instead of it being our own `.tsx` implementation.